python/ray/dag/compiled_dag_node.py

```diff
@@ -68,17 +69,41 @@ def do_exec_compiled_task(
             for idx, channel in input_channel_idxs:
                 resolved_inputs[idx] = channel.begin_read()
```

Comment:
It might be good to explicitly try-catch the channel calls so that we can differentiate between expected errors (channel closed), application code errors, and anything else that might error in this loop (most likely system bugs). The try-catch at the end can be for system errors only.

Reply:
Hmm I played around with this and deciding the semantics is tricky, so I think we should tackle this later on for productionization.